hy/contrib/hy_repr.hy

```diff
@@ -80,9 +80,9 @@
         (.rstrip (repr x) "L")
       (and (in t [float HyFloat]) (isnan x))
         "NaN"
-      (= x Inf)
+      (and (in t [float HyFloat]) (= x Inf))
         "Inf"
-      (= x -Inf)
+      (and (in t [float HyFloat]) (= x -Inf))
```

Comment:
TBH now that I'm looking at this again, it should probably be `(issubclass t float)`...